streamer/periodconcat_node.py

```diff
@@ -38,6 +41,39 @@ def __init__(self,
     self._pipeline_config = pipeline_config
     self._output_dir = output_dir
     self._packager_nodes: List[PackagerNode] = packager_nodes
+    
+    # know whether the first period has video and audio or not.
+    fp_has_vid, fp_has_aud = False, False
+    for output_stream in packager_nodes[0].output_streams:
+      if isinstance(output_stream, VideoOutputStream):
+        fp_has_vid = True
+      elif isinstance(output_stream, AudioOutputStream):
+        fp_has_aud = True
+    
+    for i, packager_node in enumerate(self._packager_nodes):
+      has_vid, has_aud = False, False
+      for output_stream in packager_node.output_streams:
+        if isinstance(output_stream, VideoOutputStream):
+          has_vid = True
+        elif isinstance(output_stream, AudioOutputStream):
+          has_aud = True
+      if has_vid != fp_has_vid or has_aud != fp_has_aud:
+        # Overwrite the start and the stop methods.
+        setattr(self, 'start', lambda: None)
+        setattr(self, 'stop', lambda _=None: None)
```

Comment:
I don't like overwriting these methods.  How about you set a flag like self._fatal_error = True and then throw a RuntimeError at the top of _thread_single_pass if it's set?

Reply:
ah, sure, 
Can we not error when we see the flag?, i think of supporting multiple inputs even if it's not going to be concatenated.
Can we just set the status to finished in the _thread_single_pass.

Reply:
We don't intend to support it in the player, so we don't intend to support that scenario here, either.  I think it would be best to guide people toward more reasonable, widely-supported structures in media, until there's a compelling use-case for us to broaden support.  We haven't heard one yet.

Reply:
What i meant is that the user can use one streamer instance to process multiple inputs, instead of instantiating multiple instances of streamer, for LIVE for example, since there is no concatenation support, i just didn't append the concater node [here](https://github.com/google/shaka-streamer/blob/master/streamer/controller_node.py#L170), so the user can push multiple live streams from the same streamer instance.

Reply:
oh i think i miss understood you, ok let's raise an error till someone requests not to.